### PR TITLE
seeding roles, activities and their map table

### PR DESCRIPTION
**Pull request by thetif**
updated the way states, roles, activities, and roles-activity-mapping are seeded; added seeding to build process

### This pull request changes...

- adds seeding to the prod build process
- only sees states if there are no states
- deletes all of the activities before re-adding them
- deletes all of the roles-activities before re-adding them
- adds logging to states and roles seeding

### This pull request is ready to merge when...

- [ ] Tests have been updated (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [ ] The change has been documented
~- [ ] Associated OpenAPI documentation has been updated~
- [ ] Changelog is updated as appropriate

### CODE REVIEW

File: api/migrations/20200921100859_change-role-names.js [hunk 1]
```diff
@@ -16,16 +16,16 @@ exports.up = async knex => {
...
 
   await knex('auth_roles')
     .where({ name: 'state coordinator' })
     .update({ name: 'eAPD State Coordinator', isActive: false });
 
   await knex('auth_roles')
-    .where({ name: 'state' })
-    .update({ name: 'eAPD State', isActive: false });
+    .where({ name: 'state SME' })
+    .update({ name: 'eAPD State SME', isActive: false });
```
Comment (radavis) [inline api/migrations/20200921100859_change-role-names.js:28]: If this migration filename appears in the `knex_migrations` table, this code will not run.

You probably want to generate a new migration and make the necessary changes to the data.

Comment (thetif) [reply api/migrations/20200921100859_change-role-names.js:28]: I already changed it in the database, the database value is "state SME" and "federal SME" and this migration originally missed it. I changed it so it would work in prod.

File: api/seeds/shared/states.js [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
 const { states } = require('../../util/states');
 
 exports.seed = async knex => {
-  await knex('states').insert(states);
+  logger.info(`Begging to seed the ${chalk.cyan('states')} table`);
+  const total = await knex('states')
+    .count('id')
+    .first();
+  logger.info(`${chalk.cyan('states')} table currently has ${total.count}`);
+  if (total.count.toString() === '0') {
```
Comment (radavis) [inline api/seeds/shared/states.js:11]: Is this right?

File: api/seeds/shared/roles-and-activities.js [hunk 3]
```diff
@@ -135,4 +177,5 @@ exports.seed = async knex => {
       'eAPD State SME'
     ])
     .update({ isActive: false });
+  logger.info('Updated active status for roles');
```
Comment (radavis) [inline api/seeds/shared/roles-and-activities.js:180]: Running the seed command on my local database yields the appropriate associations in these tables.

File: api/seeds/shared/states.js (hunk 2 above)
Comment (thetif) [reply api/seeds/shared/states.js:11]: The documentation says it comes back as a bigint or a string, so I converted it to a string just in case and tested it that way.

Comment (thetif) [reply api/seeds/shared/states.js:11]: But also when I tested it on my system it did wipe out the states and saw that it was empty and added them back without duplicating them.

**Review by radavis — APPROVED** [review]
